combat-trainer.lic

```diff
@@ -1646,8 +1625,17 @@ class SpellProcess
 
     game_state.hide_on_cast = true if data['use_stealth']
 
-    if data['runestone_for_combat_exists']
-      cast_runestone(data, game_state)
+    if data['runestone_name']
+      if DRCI.inside?(data['runestone_name'], @settings.runestone_storage)
+        data['runestone_for_combat_exists']=true
+        cast_runestone(data, game_state)
+      else
+        ### Just need this setting to remove the spell from the list
+        data['runestone_for_combat_exists']=false
+        @offensive_spells.reject! { |spell| spell['runestone_for_combat_exists']==false }
+        return
+      end
+      prepare_spell(data, game_state, @buff_force_cambrint)
```

Comment:
Possible typo?
```suggestion
      prepare_spell(data, game_state, @buff_force_cambrinth)
```

Also, Is line 1638 supposed to be here? It looks like if you have a runestone then it's cast at line 1631 and then `prepare_spell` gets called at 1638 -- is that expected?

Reply:
`prepare_spell` was there before so need to call it if not doing runestone piece. I was testing getting cambrinth for these spells since you invoke at base. The harness option works. I had a thought last night on how to get the mana option to work.

Reply:
Actually, let me try just one of these outside the if again.